src/platform_impl/linux/x11/mod.rs

```diff
@@ -392,11 +384,27 @@ impl<T: 'static> EventLoop<T> {
                     callback,
                 );
             }
+
+            // Quickly dispatch all pending events to clear channels
+            this.event_loop
+                .dispatch(
+                    Some(Duration::ZERO),
+                    &mut get_xtarget(&this.target).state.borrow_mut(),
+                )
+                .ok();
+
             // Empty the redraw requests
             {
                 let mut windows = HashSet::new();
 
-                while let Ok(window_id) = this.redraw_receiver.try_recv() {
+                // Empty the channel.
+
+                while let Some(window_id) = get_xtarget(&this.target)
+                    .state
+                    .borrow_mut()
+                    .redraw_events
+                    .pop_front()
+                {
                     windows.insert(window_id);
                 }
```

Comment:
Is the logic to push `redraw` events further here? I'd probably suggest to dispatch just redraw requests if that's the intention, so the further logic with empty checks could be eliminated.

Though, it's a matter of preference, maybe we want to dispatch all of them.

Reply:
Sounds good, I've moved the redraw channel into a `Dispatcher` to make this easier.